### ISSUE TITLE
 virtual display files not cleared, multiple users would crash

### ISSUE THREAD

**Issue by adchooser** [mozilla/OpenWPM#281]
Hi,
I installed OpenWPM on a system, where multiple users can use it. Users run it in headless mode.
After one user successfully run a test, an other user can't start it:
```
BrowserManager       - INFO     - BROWSER 28: Crash in driver, restarting browser manager 
 Traceback (most recent call last):
  File "/usr/lib/python2.7/OpenWPM/automation/BrowserManager.py", line 347, in BrowserManager
    status_queue, browser_params, manager_params, crash_recovery)
  File "/usr/lib/python2.7/OpenWPM/automation/DeployBrowsers/deploy_browser.py", line 15, in deploy_browser
    manager_params, crash_recovery)
  File "/usr/lib/python2.7/OpenWPM/automation/DeployBrowsers/deploy_firefox.py", line 161, in deploy_firefox
    firefox_options=fo, log_path=interceptor.fifo)
  File "/usr/local/lib/python2.7/dist-packages/selenium/webdriver/firefox/webdriver.py", line 154, in __init__
    keep_alive=True)
  File "/usr/local/lib/python2.7/dist-packages/selenium/webdriver/remote/webdriver.py", line 151, in __init__
    self.start_session(desired_capabilities, browser_profile)
  File "/usr/local/lib/python2.7/dist-packages/selenium/webdriver/remote/webdriver.py", line 240, in start_session
    response = self.execute(Command.NEW_SESSION, parameters)
  File "/usr/local/lib/python2.7/dist-packages/selenium/webdriver/remote/webdriver.py", line 308, in execute
    self.error_handler.check_response(response)
  File "/usr/local/lib/python2.7/dist-packages/selenium/webdriver/remote/errorhandler.py", line 194, in check_response
    raise exception_class(message, screen, stacktrace)
WebDriverException: Message: connection refused


BrowserManager       - ERROR    - BROWSER 28: Spawn unsuccessful  | Proxy Ready: False  | Profile Created: True  | Profile Tar: True  | Display: True  | Launch Attempted: True  | Browser Launched: False  | Browser Ready: False
```

I find out, that it happen because the virtualdisplay didn't cleared fully: there are files remaining in the `/tmp/.X11-unix/` directory, **owned by an other user**, and the virtualdisplay want to use these, but it don't have permission for that. After a `sudo rm /tmp/.X11-unix/*`, the problem solved (temporary).
```
$ ls -la /tmp/.X11-unix/
total 8
drwxrwxrwt  2 root   root   4096 Apr 12 07:29 .
drwxrwxrwt 12 root   root   4096 Apr 12 07:32 ..
srwxrwxrwx  1 user1 user1    0 Apr 12 07:29 X1001
srwxrwxrwx  1 user1 user1    0 Apr 12 07:29 X1005
srwxrwxrwx  1 user1 user1    0 Apr 12 07:29 X1009
```

`display.stop()` would clear these

Versions:
```
$ cat /usr/lib/python2.7/OpenWPM/VERSION 
0.8.0
```
```
$ python -c "import pyvirtualdisplay; print(pyvirtualdisplay.__version__)"
0.2.1
```
```
$ python --version
Python 2.7.15rc1
```
```
$ uname -a
Linux vm 4.15.0-47-generic #50-Ubuntu SMP Wed Mar 13 10:44:52 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by englehardt** [comment]
Thanks! This ties in to #12 as well. We should handle shutdown more gracefully, which would greatly lessen the chance of this happening. It won't fix it entirely, as crashes or keyboard interrupts may still lead to display lock files being left around. I guess a solution there could be to change the `TMPDIR` env variable to be user-specific.

**Comment by englehardt** [comment]
Will be fixed by #426

**Comment by englehardt** [comment]
Closing since we no longer use XVFB/pyvirtualdisplay (#426)
